clustermesh install documentation: missing step

dashaun opened the pull request:
As I normally follow along with documentation, I usually look at the code blocks to see which steps I need to execute.

I was starting from scratch, following the docs, and never saw a *cilium install* step. When I got to the *cilium clustermesh enable* step I got an error, because I hadn't installed Cilium into the cluster yet.

This patch simply points out that Cilium needs to be installed on each cluster, and it gives a code block, with a copy/paste example.

Please ensure your pull request adheres to the following guidelines:

- [X] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [X] All code is covered by unit and/or runtime tests where feasible.
- [X] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [X] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [X] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [ ] Provide a title or release-note blurb suitable for the release notes.
- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [ ] Thanks for contributing!

marseel commented on Documentation/network/clustermesh/clustermesh.rst:
```diff
@@ -94,6 +96,13 @@ time of Cilium:
  * Helm options ``cluster.name`` and ``cluster.id``
  * Cilium CLI install options ``--set cluster.name`` and ``--set cluster.id``
 
+Example install using the Cilium CLI:
```
I think it would make sense to link to the installation guide too https://docs.cilium.io/en/stable/gettingstarted/k8s-install-default/#cilium-quick-installation
as it captures more cases of how to install Cilium for different cloud providers etc and it probably makes sense to familiarize yourself with it before copying commands :)

dashaun replied:
"See other use cases and familiarize yourself with the Cilium CLI".  Under the code example?

The thing is, for this install, users have to set the cluster.name and cluster.id, if we send them to the quick install, they might go through the install, and the validation, without setting those values.

I wanted to link to the other docs, I just wasn't sure which link wouldn't get them off track.

Maybe the link to the quick installation should go by this sentence?

"It is best to assign both these attributes at installation time of Cilium." (link to quick install)

dashaun replied:
Added a link after the required install options, and before the code block.

marseel reviewed (changes requested):
Thanks for your contribution, definitely makes sense to me.

learnitall commented on Documentation/network/clustermesh/clustermesh.rst:
```diff
@@ -94,6 +96,15 @@ time of Cilium:
  * Helm options ``cluster.name`` and ``cluster.id``
  * Cilium CLI install options ``--set cluster.name`` and ``--set cluster.id``
 
+Review :ref:`_k8s_quick_install` for more details and use cases.
```
Looks like the Netifly build is failing because this link isn't quite right. Can you change it to `k8s_install_quick`?

```suggestion
Review :ref:`k8s_install_quick` for more details and use cases.
```

learnitall reviewed (changes requested):
There's one change needed to get a successful build. Otherwise LGTM. Thanks!

marseel reviewed (approved):
Thanks, looks good from my side.

learnitall reviewed (approved):
Thanks!

dashaun committed:
Docs: clustermesh install documentation, add missing step

As I normally follow along with documentation, I usually look at the code blocks to see which steps I need to execute.

I was starting from scratch, following the docs, and never saw a *cilium install* step.
When I got to the *cilium clustermesh enable* step I got an error, because I hadn't installed Cilium into the cluster yet.

This patch simply points out that Cilium needs to be installed on each cluster, and it gives a code block, with a copy/paste example.

Signed-off-by: DaShaun Carter <dashaun@dashaun.com>
